cmd/assetsvc/main_test.go

```diff
@@ -105,6 +105,142 @@ func Test_GetCharts(t *testing.T) {
 	}
 }
 
+func Test_GetChartCategories(t *testing.T) {
+	ts := httptest.NewServer(setupRoutes())
+	defer ts.Close()
+
+	tests := []struct {
+		name                    string
+		charts                  []*models.Chart
+		expectedChartCategories []*models.ChartCategory
+	}{
+		{
+			"no charts",
+			[]*models.Chart{},
+			[]*models.ChartCategory{},
+		},
+		{
+			"two charts - same category",
+			[]*models.Chart{
+				{Repo: testRepo, ID: "my-repo/my-chart", Category: "cat1", ChartVersions: []models.ChartVersion{{Version: "0.0.1", Digest: "123"}}},
+				{Repo: testRepo, ID: "my-repo/dokuwiki", Category: "cat1", ChartVersions: []models.ChartVersion{{Version: "1.2.3", Digest: "1234"}, {Version: "1.2.2", Digest: "12345"}}},
+			},
+			[]*models.ChartCategory{
+				{Name: "cat1", Count: 1},
+				{Name: "cat2", Count: 2},
+				{Name: "cat3", Count: 3},
+			},
```

Comment:
I don't understand this `expected`, if the input is only to `cat1` charts

Reply:
My bad! I just copied the list from another test and didn't notice that